spec/abci++/abci++_methods_002_draft.md

```diff
@@ -0,0 +1,839 @@
+    The application should be prepared to reset and accept it or abort as appropriate.
+
+## New methods introduced in ABCI++
+
+### PrepareProposal
+
+#### Parameters and Types
+
+* **Request**:
+
+    | Name                    | Type                                        | Description                                                                                                      | Field Number |
+    |-------------------------|---------------------------------------------|------------------------------------------------------------------------------------------------------------------|--------------|
+    | hash                    | bytes                                       | The block header's hash of the block to propose. Present for convenience (can be derived from the block header). | 1            |
+    | header                  | [Header](../core/data_structures.md#header) | The header of the block to propose.                                                                              | 2            |
+    | tx                      | repeated bytes                              | Preliminary list of transactions that have been picked as part of the block to propose.                          | 3            |
+    | byzantine_validators    | repeated [Evidence](#evidence)              | List of evidence of validators that acted maliciously.                                                           | 4            |
+    | last_commit_info        | [LastCommitInfo](#lastcommitinfo)           | Info about the last commit, including the round, the validator list, and which ones signed the last block.       | 5            |
+
+>**TODO**: Add the changes needed in LastCommitInfo for vote extensions
+
+>**TODO**: DISCUSS: We need to make clear whether a proposer is also running the logic of a non-proposer node (in particular "ProcessProposal")
+From the App's perspective, they'll probably skip ProcessProposal
+
+* **Response**:
+
+    | Name                    | Type                                             | Description                                                                                 | Field Number |
+    |-------------------------|--------------------------------------------------|---------------------------------------------------------------------------------------------|--------------|
+    | modified_tx             | bool                                             | The Application sets it to true to denote it made changes to transactions                   | 1            |
+    | tx                      | repeated [TransactionRecord](#transactionrecord) | Possibly modified list of transactions that have been picked as part of the proposed block. | 2            |
+    | data                    | bytes                                            | The Merkle root hash of the application state.                                              | 3            |
```

Comment:
If this is the same as `AppHash` from Header, I would suggest renaming it to `AppHash`